lib/Basics/RebootId.h

```diff
@@ -0,0 +1,97 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Tobias Gödderz
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <iosfwd>
+#include <limits>
+#include <memory>
+#include <string>
+#include "velocypack/Builder.h"
+
+namespace arangodb {
+class RebootId {
+ public:
+  using value_type = uint64_t;
+
+  explicit constexpr RebootId() noexcept = delete;
+  explicit constexpr RebootId(value_type rebootId) noexcept
+      : _value(rebootId) {}
+  [[nodiscard]] value_type value() const noexcept { return _value; }
+
+  [[nodiscard]] bool initialized() const noexcept { return value() != 0; }
+
+  [[nodiscard]] bool operator==(RebootId other) const noexcept {
+    return value() == other.value();
+  }
+  [[nodiscard]] bool operator!=(RebootId other) const noexcept {
```

Comment:
Starting from C++20 you don't required to define != if you define ==

Also I think will be better to have just two operators:
```cpp
[[nodiscard]] bool operator==(RebootId other) const noexcept = default;
[[nodiscard]] auto operator<=>(RebootId other) const noexcept = default;
```

Reply:
Well actually:
You should read [this](https://en.cppreference.com/w/cpp/language/default_comparisons) page about default comparison on cppreference, it is very informative. I learned a lot. There it says
> If operator<=> is defaulted and operator== is not declared at all, then operator== is implicitly defaulted.

so you don't have to declare `operator==`. Please also look at the following [godbolt](https://godbolt.org/z/cssxb85W8) link.

Besides that, I don't want to change the code much in this PR. Only what is necessary. I don't need it for my work, so you should go in and fix it, if you need it.